app.rb

```diff
@@ -0,0 +1,87 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    new_numbers = []
+    each do |i|
+      new_numbers.push(i) if yield(i)
+    end
+    new_numbers
+  end
+
+  def my_all?
+    condition = true
+    each do |j|
+      condition = false unless yield(j)
+    end
+    condition
+  end
+
+  def my_any?
+    each do |j|
+      return true if yield(j)
+    end
+    false
+  end
+
+  def my_none?
+    condition = true
+    each do |j|
+      condition = false if yield(j)
+    end
+    condition
+  end
```

Comment:
Method `my_none?` has very similar errors to `my_any?`, you can fix it by putting `!my_any?` inside. If `my_any?` will work correctly, `my_none?` will be correct too.